File: t01/tarea.py
from itertools import permutations

perm = permutations([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])

for i in list(perm):
    f = i[0]
    o = i[1]
    r = i[2]
    t = i[3]
    y = i[4]
    e = i[5]
    n = i[6]
    s = i[7]
    ii = i[8]
    x = i[9]

    forty = (f*10000) + (o*1000) + (r*100) + (t*10) + (y)
    sixty = (s*10000) + (ii*1000) + (x*100) + (t*10) + (y)
    ten = (t*100) + (e*10) + (n)

    if (sixty == (forty + 2*ten)):
        print("forty:", forty)
        print("sixty:", sixty)
        print("ten:", ten)

File: t01/tarea.cpp
#include <iostream>
#include <stdlib.h>
#include <time.h>

/*	FORTY
 *	  TEN
 *	  TEN
 *	-----
 *	SIXTY
 */

int main()
{
    int f, o, r, t, y, e, n, s, i, x;
    int ten, forty, sixty, suma;
    srand(time(NULL));
    do {
	do {
	    f = rand() % 10;
	    o = rand() % 10;
	    r = rand() % 10;
	    t = rand() % 10;
	    y = rand() % 10;
	    e = rand() % 10;
	    n = rand() % 10;
	    s = rand() % 10;
	    i = rand() % 10;
	    x = rand() % 10;

	    ten = (t * 100) + (e * 10) + n;
	    forty = (f * 10000) + (o * 1000) + (r * 100) + (t * 10) + y;
	    sixty = (s * 10000) + (i * 1000) + (x * 100) + (t * 10) + y;

	    suma = f + o + r + t + y + e + n + s + i + x;
	} while (((ten * 2) != (sixty - forty)) && suma != 45);
    } while (f == o && f == r && f == t && f == y && f == e && f == n && f == s && f == i && f == x && o == r && o == t && o == y && o == e && o == n && o == s && o == i && o == x);
    std::cout << "ten = " << ten
	      << "\nforty = " << forty
	      << "\nsixty = " << sixty << std::endl;
}
